src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -235,31 +235,40 @@ ExtentPlacementManager::open_for_write()
   });
 }
 
-ExtentPlacementManager::alloc_paddr_iertr::future<>
-ExtentPlacementManager::delayed_alloc_or_ool_write(
-    Transaction &t,
-    const std::list<LogicalCachedExtentRef> &delayed_extents)
+ExtentPlacementManager::dispatch_result_t
+ExtentPlacementManager::dispatch_delayed_extents(Transaction &t)
 {
-  LOG_PREFIX(ExtentPlacementManager::delayed_alloc_or_ool_write);
-  DEBUGT("start with {} delayed extents",
-         t, delayed_extents.size());
-  assert(writer_refs.size());
-  return seastar::do_with(
-      std::map<ExtentOolWriter*, std::list<LogicalCachedExtentRef>>(),
-      [this, &t, &delayed_extents](auto& alloc_map) {
-    for (auto& extent : delayed_extents) {
-      // For now, just do ool allocation for any delayed extent
+  projected_usage_t usage;
+  dispatch_result_t res;
+  auto delayed_extents = t.get_delayed_alloc_list();
+
+  for (auto& extent : delayed_extents) {
+    if (dispatch_delayed_extent(extent)) {
+      t.mark_delayed_extent_inline(extent);
+      usage.inline_usage += extent->get_length();
```

Comment:
The type of `ool_block_list` is `std::list<CachedExtentRef>`, does change it to `std::list<LogicalCachedExtentRef>` reasonable?

Reply:
out-of-line needs to be applied to physical extents with RBM as it's journaling space is statically partitioned, so it should be fine.

Reply:
delay extent is pushed into ool_block_list, so it can be passed to LBA::update_mappings directly when it's logical.

Since it can't be logical, dispatch_result_t won't change to trace all extents need to update mapping.

Reply:
I think it should be good not to worry about the ool physical extents until later.